packages/axelar-wasm-std/src/address.rs

```diff
@@ -40,6 +41,15 @@ pub fn validate_address(address: &str, format: &AddressFormat) -> Result<(), Err
             ScAddress::from_str(address)
                 .change_context(Error::InvalidAddress(address.to_string()))?;
         }
+        AddressFormat::Base58Solana => {
+            const SOLANA_PUBKEY_LEN: usize = 32;
+            let pubkey_vec = bs58::decode(address)
+                .into_vec()
+                .change_context(Error::InvalidAddress(address.to_string()))?;
+            if pubkey_vec.len() != SOLANA_PUBKEY_LEN {
+                bail!(Error::InvalidAddress(address.to_string()))
+            }
+        }
```

Comment:
I suggest doing this check using `solana_sdk::pubkey::Pubkey` to avoid manual verification. It's more secure and maintainable that way, also it's more consistent with the other addresses format verifications

Reply:
In case using the solana-sdk is not possible due to incompatibilities with cosmwasm, please consider porting [this check](https://github.com/anza-xyz/solana-sdk/blob/4e30766b8d327f0191df6490e48d9ef521956495/pubkey/src/lib.rs#L236-L238) so that verification fails early before the actual decoding

Reply:
fixed at f67f68dc75fff61a3ace73252aba876cc70b2349